File: src/lightnap-ng/src/app/core/backend-api/dtos/identity/response/supported-external-login-dto.ts
/**
 * DTO representing a supported external login option.
 */
export interface SupportedExternalLoginDto {
    /** The unique name of the external login provider. */
    providerName: string;
    /** The display name of the external login provider. */
    displayName: string;
    /** The URL to initiate the external login process. */
    loginUrl: string;
}

File: src/LightNap.WebApi/Configuration/SupportedExternalLoginDto.cs
using System.ComponentModel.DataAnnotations;

namespace LightNap.WebApi.Configuration
{
    /// <summary>
    /// Represents an external authentication provider supported by the application, including its identifier and
    /// display name.
    /// </summary>
    /// <param name="ProviderName">The unique name or identifier of the external authentication provider (for example, "Google" or "Facebook").
    /// Cannot be null or empty.</param>
    /// <param name="DisplayName">The user-friendly name of the external authentication provider as displayed in the UI. Cannot be null or empty.</param>
    public record SupportedExternalLoginDto(string ProviderName, string DisplayName)
    {
        /// <summary>
        /// Gets the full path URL for initiating an external login with the specified provider.
        /// </summary>
        public string LoginUrl => $"/api/ExternalLogin/login/{ProviderName}";
    }
}